br/cmd/br/cmd.go

```diff
@@ -162,6 +182,23 @@ func Init(cmd *cobra.Command) (err error) {
 		}
 		log.ReplaceGlobals(lg, p)
 		memory.InitMemoryHook()
+		if debug.SetMemoryLimit(-1) == math.MaxInt64 {
+			memtotal, e := memory.MemTotal()
+			if e != nil {
+				err = e
+				return
+			}
+			memused, e := memory.MemUsed()
+			if e != nil {
+				err = e
+				return
+			}
+			memleft := memtotal - memused
```

Comment:
do we need to add a safety check here to make sure `memtotal-memused` won't underflow? since we own the memory package and if might have a bug that returned memtotal is actually smaller than memused

Reply:
done. If underflow, output a warning log and skip set memory limit.